### PR TITLE
fix: describe database latency requirements

### PR DESCRIPTION
**Pull request by jawnsy**
We require low latency between Coder and the database, so update
the PostgreSQL installation guide to indicate that fact.

### CODE REVIEW

File: guides/deployments/postgres.md
Comment (deansheather) [inline guides/deployments/postgres.md:12]:
```suggestion
will deploy a [PostgreSQL database](https://www.postgresql.org/) within the
```

File: guides/deployments/postgres.md
Comment (deansheather) [inline guides/deployments/postgres.md:15]: and for reliability

File: guides/deployments/postgres.md
Comment (deansheather) [inline guides/deployments/postgres.md:16]:
```suggestion
data is not backed up and will be lost when deleting the namespace or
```

File: guides/deployments/postgres.md
Comment (deansheather) [inline guides/deployments/postgres.md:21]:
```suggestion
> ensuring that the database is within the same data center as the control plane,
```

File: guides/deployments/postgres.md
Comment (deansheather) [inline guides/deployments/postgres.md:22]:
```suggestion
> or within the same availability zone within a cloud provider.
```

File: guides/deployments/postgres.md
Comment (deansheather) [inline guides/deployments/postgres.md:105]:
```suggestion
Once this process is completed, you'll be able to access Coder using the
```

File: guides/deployments/postgres.md
Comment (deansheather) [inline guides/deployments/postgres.md:14]:
```suggestion
however, we **recommend using an out-of-cluster database for production** to
```

File: guides/deployments/postgres.md
Comment (jawnsy) [reply guides/deployments/postgres.md:15]: I don't think we want to imply that running postgres in our namespace is _unreliable_, do we?

File: guides/deployments/postgres.md
Comment (jawnsy) [reply guides/deployments/postgres.md:12]: fixed, thank you

File: guides/deployments/postgres.md
Comment (jawnsy) [reply guides/deployments/postgres.md:16]: "database data" sounds funny to me so I went with "database state"

File: guides/deployments/postgres.md
Comment (jawnsy) [reply guides/deployments/postgres.md:21]: fixed, thanks!

File: guides/deployments/postgres.md
Comment (jawnsy) [reply guides/deployments/postgres.md]: I used this phrasing to distinguish between deploying within the same region, but not the same AZ. I think "same availability zone" is understood to mean "within the same cloud provider", no?

Most of the clouds use "availability zone" to refer to a specific failure domain (e.g. power supplies, building, etc) and we would typically consider this to be the same data center. There are sometimes multiple availability zones that are different buildings in close proximity within a region, so latency is still low in that case.

I think as long as users can guarantee low latency, it doesn't matter where the database is, so we shouldn't be prescriptive that this _must_ be in the same AZ.

File: guides/deployments/postgres.md
Comment (jawnsy) [reply guides/deployments/postgres.md:108]: Say no to passive sentences, IMO :no_good: - will defer to @khorne3 on this one

File: guides/deployments/postgres.md
Comment (jawnsy) [reply guides/deployments/postgres.md:14]: fixed, ty

File: guides/deployments/postgres.md
Comment (khorne3) [reply guides/deployments/postgres.md:108]: Yep, active voice. Thanks!

**Review by khorne3 — APPROVED** [review]

File: guides/deployments/postgres.md
Comment (deansheather) [inline guides/deployments/postgres.md:23]:
```suggestion
> plane, such as within the same cloud availability zone.
```

File: guides/deployments/postgres.md
Comment (deansheather) [inline guides/deployments/postgres.md:75]:
```suggestion
prior steps [in your Helm values file](../admin/helm-charts.md).
```

File: guides/deployments/postgres.md
Comment (deansheather) [inline guides/deployments/postgres.md:76]:
```suggestion
Coder will use these credentials to connect to your
```

File: guides/deployments/postgres.md
Comment (jawnsy) [reply guides/deployments/postgres.md:23]: Isn't it useful to have some sort of criteria so that people know what "a lot" looks like?

File: guides/deployments/postgres.md
Comment (jawnsy) [reply guides/deployments/postgres.md]: fixed

File: guides/deployments/postgres.md
Comment (jawnsy) [reply guides/deployments/postgres.md]: fixed

File: guides/deployments/postgres.md
Comment (deansheather) [reply guides/deployments/postgres.md:23]: We don't give any instructions for how to calculate "latency under load" so we should avoid giving them numbers like this. I think the rest of the paragraph does a good enough job of conveying that the latency should be in the low milliseconds or sub-millisecond range.

Comment (jawnsy) [reply guides/deployments/postgres.md:23]: I dunno, that seems like a little too read-between-the-lines thing to me, but I don't feel strongly enough about this to hold up this PR, will remove that line

**Review by deansheather — APPROVED** [review]

**Review by ericpaulsen — APPROVED** [review]

**Pull request merged**
